CMakeLists.txt

```diff
@@ -102,9 +102,19 @@ elseif(EVHTP_ALLOCATOR STREQUAL "tcmalloc")
     endif()
 endif()
 
-add_library(evhtp ${LIBEVHTP_SOURCE_FILES})
+if(EVHTP_BUILD_SHARED)
+	set (EVHTP_LIBTYPE SHARED)
+else()
+	set (EVHTP_LIBTYPE STATIC)
+endif()
+
+add_library(evhtp ${EVHTP_LIBTYPE} ${LIBEVHTP_SOURCE_FILES})
 target_link_libraries(evhtp PUBLIC ${LIBEVHTP_EXTERNAL_LIBS})
 
+if(EVHTP_BUILD_SHARED)
+	set_target_properties(evhtp PROPERTIES VERSION "${PROJECT_VERSION}" SOVERSION 0 OUTPUT_NAME "evhtp")
+endif()
+
```

Comment:
This should all be covered by passing `-DBUILD_SHARED_LIBS=ON` to the cmake options. The Hunter devs schooled me on that.

Reply:
Thanks, I tried compiling with this option last week, but not work. I tried it again just now, it's OK.